File: seminar_1/task_1/Ishmuratov.Rail/Task_1.cpp
#include <bits/stdc++.h>

using namespace std;

struct ListNode {
    int val;
    ListNode *next;
    ListNode() : val(0), next(nullptr) {}
    ListNode(int x) : val(x), next(nullptr) {}
    ListNode(int x, ListNode *next) : val(x), next(next) {}
};

ListNode* removeElements(ListNode* head, int val) {
    if(head == nullptr) {
        return head;
    }
    ListNode* curr = head;
    ListNode* prev = nullptr;
    while(curr!=nullptr){
        if(curr->val==val){
            prev->next = curr->next;
            curr = curr->next;
        }
        else{
            prev = curr;
            curr = curr->next;
        }
    }
    return head;
}

int main()
{
    ListNode d{4};
    ListNode c{3, &d};
    ListNode b{2, &c};
    ListNode a{1, &b}; // Это head
    ListNode* head = &a;
    while (head != nullptr) {
        cout<<head->val<<" ";
        head = head->next;
    }
    cout<<"\n";
    ListNode* result = removeElements(&a, 2);
    while (result != nullptr) {
        cout<<result->val<<" ";
        result = result->next;
    }
    cout<<"\n";
}

File: seminar_1/task_1/patokova.victoria/task-1.py
class ListNode:
    def __init__(self, data):
        self.data = data
        self.next = None


def delElem(head, val):
    dummy = ListNode(None)
    dummy.next = head
    prev = dummy
    cur = head

    while cur != None:
        if cur.data == val:
            prev.next = cur.next

        else:
            prev = cur

        cur = cur.next

    head = dummy.next
    del(dummy)

    return head


def printLinkedList(head):
    cur = head
    while cur != None:
        print(cur.data)
        cur = cur.next

        
e1 = ListNode(1)
e2 = ListNode(2)
e3 = ListNode(3)
e4 = ListNode(4)
e5 = ListNode(3)

e1.next = e2
e2.next = e3
e3.next = e4
e4.next = e5

head = e1

head = delElem(head, 3)
printLinkedList(head)